magellan-library/src/main/java/com/wealthfront/magellan/navigation/CurrentNavigableProvider.kt

```diff
@@ -2,35 +2,28 @@ package com.wealthfront.magellan.navigation
 
 import android.content.Context
 import com.wealthfront.magellan.OpenForMocking
-import com.wealthfront.magellan.coroutines.CreatedLifecycleScope
 import com.wealthfront.magellan.lifecycle.LifecycleAwareComponent
-import com.wealthfront.magellan.lifecycle.lifecycle
-import com.wealthfront.magellan.navigation.NavigationLifecycleEvent.NavigatedTo
-import kotlinx.coroutines.flow.collect
-import kotlinx.coroutines.flow.filterIsInstance
-import kotlinx.coroutines.launch
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
 @OpenForMocking
-public class CurrentNavigableProvider @Inject constructor(
-  createdScope: CreatedLifecycleScope
-) : LifecycleAwareComponent() {
+public class CurrentNavigableProvider @Inject constructor() : NavigationListener, LifecycleAwareComponent() {
```

Comment:
nit: probably can make this just `LifecycleAware` now

Reply:
done